compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -379,52 +382,69 @@ fn extend_type_not_partial_eq<'tcx>(
         adts_without_partialeq: FxHashSet<Span>,
         /// The user has written `impl PartialEq for Ty` which means it's non-structual,
         /// but we don't have a span to point at, so we'll just add them as a `note`.
-        manual: Vec<Ty<'tcx>>,
+        manual: FxHashSet<Ty<'tcx>>,
         /// The type has no `PartialEq` implementation, neither manual or derived, but
         /// we don't have a span to point at, so we'll just add them as a `note`.
-        without: Vec<Ty<'tcx>>,
+        without: FxHashSet<Ty<'tcx>>,
+        /// If any of the subtypes has escaping bounds (`for<'a>`) or involves a trait object, we
+        /// won't emit a note.
+        should_note: bool,
```

Comment:
sorry, one final nit, instead of setting `should_note` and then ignoring everything, wouldn't it be easier to return `ControlFlow::Break(())` and change the caller to
`if v.visit_ty(ty).is_break() { return }`

Reply:
Done